frontend/src/pages/TermPlanner/ContextMenu/ContextMenu.jsx

```diff
@@ -41,21 +45,92 @@ const ContextMenu = ({ code, plannedFor }) => {
     dispatch(addTab(code));
   };
 
+  const [isEditMarkVisible, setIsEditMarkVisible] = useState(false);
+
+  const showEditMark = () => {
+    setIsEditMarkVisible(true);
+  };
+
+  const validLetterGrades = ["SY", "PS", "CR", "DN", "HD"];
+  const [markInputBuf, setMarkInputBuf] = useState(
+    useSelector((state) => state.planner.courses[code].mark),
+  );
+
+  const handleConfirmEditMark = (mark) => {
+    const attemptedMark = (` ${mark}`).slice(1).replace(" ", "");
```

Comment:
I'm a bit confused here. Why is `mark` contained in a template literal string with a space and then sliced? Couldn't we just alter `mark` directly?

Reply:
To create a new string and avoid mutating the buffer set by `useState`.

Reply:
The `slice` here is redundant if we just use `replaceAll` (which is the preferred behaviour i wanted anyways) so it's been removed to just ```jsx
const attemptedMark = (` ${mark}).replaceAll(" ", "");
```